File: Backend/Server/Server/DTOs/UserProfileDto.cs
using Server.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Server.DTOs
{
    public class UserProfileDto
    {
        public string? Username { get; set; }
        public string? Email { get; set; }
        public string? Name { get; set; }
        public string? Surname { get; set; }
        public DateTime? DateOfBirth { get; set; }
        public string? Address { get; set; }
        public string? Type { get; set; }
        public UserStatus? Status { get; set; }
    }
}

File: Frontend/src/app/_interfaces/user-profile-dto.ts
import { UserStatus } from "../entities/enums/user-status.enum";

export interface UserProfileDto {
    username: string;
    email: string;
    name: string;
    surname: string;
    dateOfBirth: Date;
    address: string;
    type: string;
    status: string;
}
